### PR TITLE
Use py.test to tag API tests into tiers.

### PR DESCRIPTION
**Pull request by omaciel**
* Added `pytest` as a dependency in `setup.py`
* Created 3 new alias decorators to represent different tiers:
  * t1: CRUD
  * t2: Associations
  * t3: System integration / Long-running tests
* Added decorators to all existing, non-stubbed, test cases

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
         'numpy',
         'paramiko',
         'pygal',
+        'pytest',
```
Comment (elyezer) [inline setup.py]: :+1:

**Comment by elyezer** [comment]
ACK, would be good to choose a sample module and run one of the tiers, after that paste the results.

**Comment by oshtaier** [comment]
Yeah. Definitely we need few examples of test runs

**Comment by omaciel** [comment]
```bash
(robottelo)┌[omaciel☮Ogs-MacBook-Pro.local]-(~/hacking/robottelo)-[git://pytest ✔]-
└> py.test -v tests/foreman/api/test_architecture.py -m t1
==================================== test session starts ====================================
platform darwin -- Python 2.7.10, pytest-2.8.3, py-1.4.30, pluggy-0.3.1 -- /Users/omaciel/.virtualenvs/robottelo/bin/python
cachedir: .cache
rootdir: /Users/omaciel/hacking/robottelo, inifile:
collected 7 items

tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_negative_create_name PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_negative_update_name PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_positive_create_name PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_positive_delete PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_positive_update_name PASSED

============================== 2 tests deselected by "-m 't1'" ==============================
========================== 5 passed, 2 deselected in 21.06 seconds ==========================
(robottelo)┌[omaciel☮Ogs-MacBook-Pro.local]-(~/hacking/robottelo)-[git://pytest ✔]-
└> py.test -v tests/foreman/api/test_architecture.py -m t2
======================================== test session starts ========================================
platform darwin -- Python 2.7.10, pytest-2.8.3, py-1.4.30, pluggy-0.3.1 -- /Users/omaciel/.virtualenvs/robottelo/bin/python
cachedir: .cache
rootdir: /Users/omaciel/hacking/robottelo, inifile:
collected 7 items

tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_associate_with_os PASSED
tests/foreman/api/test_architecture.py::ArchitectureTestCase::test_post_hash <- robottelo/decorators.py SKIPPED

================================== 5 tests deselected by "-m 't2'" ==================================
========================= 1 passed, 1 skipped, 5 deselected in 2.19 seconds =========================
(robottelo)┌[omaciel☮Ogs-MacBook-Pro.local]-(~/hacking/robottelo)-[git://pytest ✔]-
```

**Comment by omaciel** [comment]
@oshtaier @elyezer here you go! :)

**Comment by sthirugn** [comment]
:+1:

**Comment by sthirugn** [comment]
Actually I was thinking if we can add t1, t2, t3 in `TestCase` class itself so we dont need to import them everytime?

so may be just call `@TestCase.t1` or `@UITestCase.t1` or `@CLITestCase.t1`.  By this way there is no need to import a new decorator everytime when a new tier is added. Thoughts?

**Comment by oshtaier** [comment]
and maybe, we should rename `t1` on `tier1` and etc as it is not intuitive name at all. +one letter names is a bad style in general

**Comment by omaciel** [comment]
@sthirugn so in the base test class do something like:

```python
from robottelo.decorators import t1, t2, t3

class TestCase(unittest2.TestCase):
    """Robottelo test case"""

    @classmethod
    def setUpClass(cls):  # noqa
        super(TestCase, cls).setUpClass()
        if not settings.configured:
            settings.configure()
        cls.logger = logging.getLogger('robottelo')
        # NOTE: longMessage defaults to True in Python 3.1 and above
        cls.longMessage = True
        cls.t1 = t1
        cls.t2 = t2
        cls.t3 = t3
```

Then use it as 

```python
@self.t1
```

Something like this?

**Comment by omaciel** [comment]
@oshtaier I can certainly rename them to `tierN`

**Comment by omaciel** [comment]
@elyezer what do you think of @sthirugn proposal? I'm not sure...

**Comment by elyezer** [comment]
@omaciel @sthirugn `@self.tier1` will not work since at import time there is no instance yet. I prefer import tier1 and use `@tier1` than have to write the baseclass name or the class name then tier1.

While we save one import line we add much overhead on what needs to be typed. Also every test case class will have it's "own" tier, for example UITestCase.tier1 and CLITestCase.tier1 people reading the code may thing they are different things...

**Comment by sthirugn** [comment]
@elyezer to clarify, my recommendation was to add t1, t2, t3 as class attribtues of `TestCase` and not instance attribtutes as shown in @omaciel's code.

```py
class TestCase(unittest2.TestCase):
    """Robottelo test case"""

    t1 = pytest.mark.t1
    t2 = pytest.mark.t2
    t3 = pytest.mark.t3
    # add more tiers
    # add more tiers


    @classmethod
    def setUpClass(cls):  # noqa
    ...
    ...
```

Also my proposal was to add t1, t2, t3, etc in `TestCase` only.  So UITestCase.t1 or CLITestCase.t1 or APITestCase.t1 means the same t1 in `TestCase`.

**Comment by elyezer** [comment]
@sthirugn they have the same effect as will end up as being class attributes.

I got the idea, but I prefer the consistency of importing from decorators and having all over the source code `@tier1` instead `@SomeClass.tier1`.

**Comment by sthirugn** [comment]
@elyezer: :+1: ok for me.

**Comment by omaciel** [comment]
@sthirugn @oshtaier I have renamed the decorators from `tN` to `tierN` as suggested. The flags/marks themselves, however, when invoking the tests will remain as `tN` (e.g. `py.test -v tests/foreman/api/ -m t2`). Do you also want the marks to be renamed?

**Comment by sthirugn** [comment]
@omaciel I am ok with leaving it as such.  pylint doesn't complaint about it :)

**Comment by sthirugn** [comment]
ACK

**Comment by abalakh** [comment]
I would suggest to rename the marks too. It's not obvious from user perspective:
When i mark the test with `@tier1`, I need to execute py.test with `-m t1`. Why `t1` then? I would probably never guess the right format without looking through the code.
It would be ok  to shorten the name if we marked our tests with some `@tier_very_long_name_here_1`, but `tier` isn't that long word :)
ACK pending non-blocking suggestion.

**Comment by sghai** [comment]
Yeah.. good to change marks to `tierN` to avoid any confusion. Otherwise looks great. thanks

**Comment by elyezer** [comment]
@omaciel I think the marks should be renamed for consistency.

    @sthirugn @oshtaier I have renamed the decorators from tN to tierN as suggested. The flags/marks themselves, however, when invoking the tests will remain as tN (e.g. py.test -v tests/foreman/api/ -m t2). Do you also want the marks to be renamed?

**Comment by elyezer** [comment]
APT

**Comment by omaciel** [comment]
@elyezer @sghai @abalakh @oshtaier done :)

**Comment by sghai** [comment]
:+1:

**Pull request merged**